### PR TITLE
Editor: Listing & form improvements

### LINKED ISSUE #708: Editor: Enum dropdown fixes and enhancements
Handling string attributes with enumerator has flaws as it is not used much. Here's a list of issues and possible enhancements, that should also cover implementation of https://github.com/ynput/ayon-backend/issues/313

- [ ] In the table, the value is shown instead of the label:

image

- [ ] Label for inherited value is not shown at all:

image

- [ ] "Clear to no value" does not make sense as the value always need to be one of the enum list or null. Currently it tries to set empty list (the same logic as for `list_of_string` attributes), which cause errors. IMO we could remove this button for single value dropdown completely.

image

Additional `color` and `icon` can be used with https://github.com/ynput/ayon-backend/pull/314 In that case, dropdowns could be rendered similarly to status dropdowns.

- [ ] Create an enhanced dropdown that supports icons and colored backgrounds.

Custom dropdown examples:
InstallerSelector: https://github.com/ynput/ayon-frontend/blob/2e6e030e67474ec3fe8a5b6477bf45aed5f33172/src/pages/SettingsPage/Bundles/InstallerSelector.jsx
VariantSelect: https://github.com/ynput/ayon-frontend/blob/b678ff3fd9dc4715f415e6a90be08bdc12c1f776/src/containers/AddonSettings/VariantSelector.jsx
StatusSelect: https://github.com/ynput/ayon-frontend/blob/f24b1a52ce5cab800942d2be2f8ced078021fe15/src/components/status/statusSelect.jsx

### PR DESCRIPTION
**Pull request by flynput**
"None" placeholder in editor listing when no tools are selected.
Refactoring formatAttributes and converted to TS.
Removed clear all from single-select enum fields in Editor details panel.

Closes #708

### CODE REVIEW

**Review by Innders — CHANGES_REQUESTED** [review]
Tools only has an option to clear to empty list now when it needs to be able to clear to null so it can be inherited.

_PR_
image
_current (except that tooltip is wrong because that's clear to inherit not empty list)_
image


It works when I choose "a (inherited)" value but then I don't see the inherited value.

https://github.com/user-attachments/assets/0818939d-e58f-468f-990a-dbfd168f5d65

File: src/pages/EditorPage/EditorPanel.jsx [hunk 1]
```diff
@@ -644,36 +677,60 @@ const EditorPanel = ({
...
+                    }
+                    if (!isMultiSelect) {
+                      return (
+                        'inherited - ' + attrib.enum.filter((el) => el.value === parentValue)[0]?.label || null
+                      )
+                    }
+                    const labels = attrib.enum
+                      ?.filter((el) => parentValue.includes(el.value))
+                      .map((el) => el.label || el.value)
+                    return `inhreited - (${labels.join(', ')})`
```
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:692]: type

Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:692]: typo `inherited`

File: src/pages/EditorPage/EditorPanel.jsx
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:1]: When selecting "inherited" value the dropdown should close straight away. It does not make sense to continue selecting more options.

image

And trying to save this selection causes an error.

image

Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:1]: If I set inherited and then set a different value then the inherited value will still show.

https://github.com/user-attachments/assets/876c159d-4457-47ce-8b10-acea6c054844

File: src/pages/EditorPage/EditorPanel.jsx [hunk 2]
```diff
@@ -644,36 +677,60 @@ const EditorPanel = ({
...
+                  if (inheritedOptionLabel) {
+                    placeholder = inheritedOptionLabel
+                    options = [{ value: null, label: inheritedOptionLabel }, ...attrib.enum]
+                  }
+
                   // never show value when inherited, just show placeholder
                   if (!isOwn) enumValue = null
-
                   input = (
                     <Dropdown
```
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:712]: Overflow styling issue. Maybe it's an issue ARC?

image

File: src/pages/EditorPage/EditorPanel.jsx [hunk 3]
```diff
@@ -644,36 +677,60 @@ const EditorPanel = ({
...
-                      style={{ flexGrow: 1 }}
+                      style={{
+                        flexGrow: 1,
+                        fontStyle: isOwn ? 'normal' : 'italic',
+                        color: isChanged
+                          ? 'var(--color-on-changed)'
+                          : !isOwn
+                          ? 'var(--md-ref-palette-neutral-variant60'
```
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:719]: use `--md-sys-color-outline` instead

**Review by Innders — CHANGES_REQUESTED** [review]
Getting there but still have some major bugs to clean up.

**Comment by flynput** [comment]
Code rewritten, PR open on new branch, closing this one.

**Pull request closed without merging**